src/k8s-engine/lima.ts

```diff
@@ -1365,96 +1431,99 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
+              while (true) {
+                if (this.currentAction !== Action.STARTING) {
+                  // User aborted
+                  return;
+                }
+                commandArgs = ['shell', '--workdir=.', MACHINE_NAME, 'ls', '/etc/rancher/k3s/k3s.yaml'];
+                this.lastCommand = `limactl ${ commandArgs.join(' ') }`;
+                try {
+                  let args = ['shell', '--workdir=.', MACHINE_NAME,
+                    'ls', '/etc/rancher/k3s/k3s.yaml'];
+
+                  args = this.debug ? ['--debug'].concat(args) : args;
+                  await childProcess.spawnFile(this.limactl, args,
+                    { env: this.limaEnv, stdio: 'ignore' });
+                  break;
+                } catch (ex) {
+                  console.log('Configuration /etc/rancher/k3s/k3s.yaml not present in lima vm; will check again...');
+                  await util.promisify(setTimeout)(1_000);
+                }
               }
+              console.debug('/etc/rancher/k3s/k3s.yaml is ready.');
             }
-            console.debug('/etc/rancher/k3s/k3s.yaml is ready.');
-          }
-        );
-        commandArgs = ['shell', '--workdir=.', MACHINE_NAME, 'sudo', 'cat', '/etc/rancher/k3s/k3s.yaml'];
-        this.lastCommandComment = 'Updating kubeconfig';
-        await this.progressTracker.action(
-          this.lastCommandComment,
-          50,
-          this.k3sHelper.updateKubeconfig(
-            () => this.limaWithCapture(...commandArgs)));
-
-        const client = this.client = new K8s.Client();
-
-        this.lastCommandComment = 'Waiting for services';
-        await this.progressTracker.action(
-          this.lastCommandComment,
-          50,
-          async() => {
-            await client.waitForServiceWatcher();
-            client.on('service-changed', (services) => {
-              this.emit('service-changed', services);
-            });
-          }
-        );
+          );
+          commandArgs = ['shell', '--workdir=.', MACHINE_NAME, 'sudo', 'cat', '/etc/rancher/k3s/k3s.yaml'];
+          this.lastCommandComment = 'Updating kubeconfig';
+          await this.progressTracker.action(
+            this.lastCommandComment,
+            50,
+            this.k3sHelper.updateKubeconfig(
+              () => this.limaWithCapture(...commandArgs)));
 
-        this.activeVersion = desiredVersion;
-        this.currentPort = this.#desiredPort;
-        this.emit('current-port-changed', this.currentPort);
+          this.client = new K8s.Client();
 
-        // Remove traefik if necessary.
-        if (!this.cfg?.options.traefik) {
+          this.lastCommandComment = 'Waiting for services';
           await this.progressTracker.action(
-            'Removing Traefik',
+            this.lastCommandComment,
             50,
-            this.k3sHelper.uninstallTraefik(this.client));
-        }
+            async() => {
+              this.client = new K8s.Client();
```

Comment:
Making a _second_ client seems very wrong…

Reply:
Sure. I was arguing with typescript here. Will cast the inner references to avoid the double creation.